File: rectangleArea.cpp
#include <iostream>
using namespace std;
///start program///

//method to prompt and obtain input from the user
void getValues(double &l, double &w)
{
    cout<<"Enter the length of the Rectangle"<<endl;
    cin>>l;
    
    cout<<"Enter the width of the Rectangle"<<endl;
    cin>>w;
}

// method to calculate the area of the rectangle
double findArea(double length, double width)
{
    return length * width;
}
// method overloading to calculate area of rectangle with different data types
double findArea(int length, int width)
{
    return length * width;
}

double findArea(int length, double width)
{
    return length * width;
}

double findArea(double length, int width)
{
    return length * width;
}

// main method
int main() 
{
    // declare variables
  double length, width, area;
  string sentinel;
  
  do {
  // obtain values
  getValues(length, width);
  
  //calculate area
  
  area = findArea(length,width);
  
  //display the area
  cout<<area<<endl;
  // ask user if they want to calculate another rectangle
  cout<<"\n \n Do you want to calculate another rectangle?\n(y/n)"<<endl;
  // obtain user input for sentinel
  cin>>sentinel;
  
  }
while (sentinel =="y");// loop will continue as long as user enters "y"
    return 0;
}

File: rectangleArea.py
#area function to calculate the area of a rectangle
'''This function takes the length and width of a rectangle as input parameters, calculates the area by multiplying them together, and returns the result. The area is calculated using the formula: Area = Length x Width.
Parameters:
- l: The length of the rectangle.
- w: The width of the rectangle.
Returns:- The area of the rectangle calculated as the product of length and width.
''' 
def area(l,w):
    area = l* w
    return area
    
#getdoubleInRange function to validate user input
'''This function prompts the user for a double value within a specified range (low to high). It continues to prompt until a valid input is received. If the input is not a valid number or is outside the specified range, it displays an error message and prompts again.
Parameters:
- prompt: A string message to display to the user when asking for input.    
- low: The lower bound of the valid range for the input.
- high: The upper bound of the valid range for the input.
Returns:- A valid double value entered by the user that falls within the specified range.'''
def getdoubleInRange(prompt, low, high):
    while True:
        try:
            value = float(input(prompt))
            if low <= value <= high:
                return value
            else:
                print(f"Please enter a number between {low} and {high}.")
        except ValueError:
            print("Invalid input. Please enter a valid number.")

#main function to execute the program
def main():
     # Define prompts for user input for length and width of the rectangle, specifying the valid range (between 5 and 10).
    lenghtPrompt = "Enter the length of the rectangle (between 5 and 10): "
    widthPrompt = "Enter the width of the rectangle (between 5 and 10): "

        # Call the getdoubleInRange function to get valid length and width values from the user, ensuring they are within the specified range of 5 to 10.
    lenght = getdoubleInRange(lenghtPrompt, 5, 10)

    # Call the getdoubleInRange function again to get the width from the user, ensuring it is also within the specified range of 5 to 10.
    width = getdoubleInRange(widthPrompt, 5, 10)

        # Call the area function to calculate the area of the rectangle using the length and width provided by the user, and store the result in the variable Rec_area.
    Rec_area = area(lenght, width)
    
   # Display the calculated area to the user with a descriptive message.
    print( "The area of the rectangle is ",Rec_area)
    
main()